src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/staves/FireStaff.java

```diff
@@ -0,0 +1,124 @@
+        super(group, item, recipeType, recipe);
+
+        setMaxUseCount(MAX_USES);
+    }
+
+    @Override
+    protected @Nonnull
+    NamespacedKey getStorageKey() {
+        return usageKey;
+    }
+
+    @Nonnull
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            Player p = e.getPlayer();
+            ItemStack item = e.getItem();
+            Location loc = p.getTargetBlock(null, 7).getLocation();
+            Block block = loc.getBlock();
+
+            if (p.getFoodLevel() >= 4 || p.getGameMode() == GameMode.CREATIVE) {
+                // Get a target block with max. 7 blocks of distance
+                if (loc.getWorld() != null && loc.getChunk().isLoaded()) {
+                    if (loc.getWorld().getPVP() && Slimefun.getProtectionManager().hasPermission(p, loc, Interaction.PLACE_BLOCK)) {
+                        e.cancel();
+                        useItem(p,item,block,loc);
+                    } else {
+                        Slimefun.getLocalization().sendMessage(p, "messages.cannot-place", true);
+                    }
+                }
+            } else {
+                Slimefun.getLocalization().sendMessage(p, "messages.hungry", true);
+            }
+        };
+    }
+
+    @ParametersAreNonnullByDefault
+    private void useItem(Player p, ItemStack item, Block block, Location loc) {
+
+
+        if (!p.isSneaking()) {
+            if (block.getType() == Material.WATER) {
+                block.setType(Material.AIR);
+                p.playSound(loc, Sound.BLOCK_FIRE_EXTINGUISH, 1, 1);
+            }
+
+            if (block.getRelative(BlockFace.UP).getType().isAir()) {
+                block.getRelative(BlockFace.UP).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
+
+            if (block.getRelative(BlockFace.DOWN).getType().isAir()) {
+                block.getRelative(BlockFace.DOWN).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
+
+            if (block.getRelative(BlockFace.EAST).getType().isAir()) {
+                block.getRelative(BlockFace.EAST).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
+
+            if (block.getRelative(BlockFace.SOUTH).getType().isAir()) {
+                block.getRelative(BlockFace.SOUTH).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
+
+            if (block.getRelative(BlockFace.NORTH).getType().isAir()) {
+                block.getRelative(BlockFace.NORTH).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
+
+            if (block.getRelative(BlockFace.WEST).getType().isAir()) {
+                block.getRelative(BlockFace.WEST).setType(Material.FIRE);
+                p.playSound(loc, Sound.ITEM_FLINTANDSTEEL_USE,1,1);
+            }
```

Comment:
*internal screaming*


Methods: Am I a joke to you?

Reply:
im sorry 😥